### PR TITLE
[Enhancement](jdbc catalog) Support reading some array types of PostgreSQL

### PR DESCRIPTION
**Pull request by zy-kkk**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

This PR supports reading some PostgreSQL array types, including:

```
{
                "int2", "int4", "int8", "smallserial", "serial",
                "bigserial", "float4", "float8", "timestamp", "timestamptz",
                "date", "bool", "bpchar", "varchar", "text"
        }
```

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcPostgreSQLClient.java [hunk 1]
```diff
@@ -99,8 +151,43 @@ protected Type jdbcTypeToDoris(JdbcFieldSchema fieldSchema) {
...
+    }
+
+    private Type convertArrayType(JdbcFieldSchema fieldSchema) {
+        int arrayDimensions = fieldSchema.getArrayDimensions().orElse(0);
+        if (arrayDimensions == 0) {
+            return Type.UNSUPPORTED;
+        }
+
+        String innerType = fieldSchema.getDataTypeName().orElse("unknown").substring(1);
+        String[] supportedInnerType = new String[] {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcPostgreSQLClient.java:171]: Make it is static final field

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**